server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteRestoreSnapshotIT.java

```diff
@@ -746,4 +754,283 @@ public void testInvalidRestoreRequestScenarios() throws Exception {
         assertTrue(exception.getMessage().contains("cannot remove setting [index.remote_store.segment.repository]" + " on restore"));
     }
 
+    public void testCreateSnapshotV2() throws Exception {
+
+        internalCluster().startClusterManagerOnlyNode();
+        internalCluster().startDataOnlyNode();
+        internalCluster().startDataOnlyNode();
+        String indexName1 = "testindex1";
+        String indexName2 = "testindex2";
+        String indexName3 = "testindex3";
+        String snapshotRepoName = "test-create-snapshot-repo";
+        String snapshotName1 = "test-create-snapshot1";
+        Path absolutePath1 = randomRepoPath().toAbsolutePath();
+        logger.info("Snapshot Path [{}]", absolutePath1);
+
+        assertAcked(
+            client().admin()
+                .cluster()
+                .preparePutRepository(snapshotRepoName)
+                .setType(FsRepository.TYPE)
+                .setSettings(
+                    Settings.builder()
+                        .put(FsRepository.LOCATION_SETTING.getKey(), absolutePath1)
+                        .put(FsRepository.COMPRESS_SETTING.getKey(), randomBoolean())
+                        .put(FsRepository.CHUNK_SIZE_SETTING.getKey(), randomIntBetween(100, 1000), ByteSizeUnit.BYTES)
+                        .put(BlobStoreRepository.REMOTE_STORE_INDEX_SHALLOW_COPY.getKey(), true)
```

Comment:
1. can we add tests with  `REMOTE_STORE_INDEX_SHALLOW_COPY` as false ? 
2. Assert that timestamp is actually pinned in remote repo on success ?

Reply:
> 1. can we add tests with  `REMOTE_STORE_INDEX_SHALLOW_COPY` as false ?
Added a test with `REMOTE_STORE_INDEX_SHALLOW_COPY` as false with enabled snapshot v2 setting
> 2. Assert that timestamp is actually pinned in remote repo on success ?
Yeah planning to do so once we have a method in RemotePinnedTimestampService service to support it. Can take in follow up PR

Reply:
Resolving the comment for now. Please reopen if needed